x-pack/plugins/cloud_security_posture/server/routes/configuration/update_rules_configuration.ts

```diff
@@ -77,57 +69,197 @@ const getEnabledRulesByBenchmark = (rules: SavedObjectsFindResponse<CspRule>['sa
+ */
+const getRuleUpdateFields = ({
+  id,
+  type,
+  attributes,
+  references,
+}: SavedObject<SavedObjectRuleUpdatePayload>) => ({ id, type, attributes, references });
+
+/**
+ * Gets current rules with next values for 'enabled' property
+ */
+const getNextCspRules = (
+  current: Array<SavedObject<SavedObjectRuleUpdatePayload>>,
+  rulesToChange: UpdateRulesConfigBodySchema['rules']
+) => {
+  const currentRulesMap = Object.fromEntries(
+    current.map((rule) => [rule.id, getRuleUpdateFields(rule)])
+  );
+  return rulesToChange.map((rule) => ({
+    ...currentRulesMap[rule.id],
+    attributes: {
+      ...currentRulesMap[rule.id].attributes,
+      enabled: rule.enabled,
+    },
+  }));
+};
+
+/**
+ * Combines updating rules in saved-objects and package policy vars into a single operation
+ * @internal
+ */
+export const updateRulesById = async (
+  { soClient, packagePolicyService, esClient, user, logger }: CspApiRequestHandlerContext,
+  { rules: rulesToChange, package_policy_id: packagePolicyId }: UpdateRulesConfigBodySchema
+): Promise<PackagePolicy> => {
+  logger.info(`Start updating rules: ${JSON.stringify(rulesToChange)}`);
+
+  const currentRules = (
+    await getPartialCspRulesById(
+      soClient,
+      rulesToChange.map((rule) => rule.id)
+    )
+  ).saved_objects;
+
+  await soClient.bulkUpdate(getNextCspRules(currentRules, rulesToChange));
+
+  try {
+    const packagePolicy = await packagePolicyService.get(soClient, packagePolicyId);
+    if (!packagePolicy) throw new Error(`Missing package policy - ${packagePolicyId}`);
+
+    const updatedPolicy = await updatePackagePolicyVars({
+      soClient,
+      packagePolicyService,
+      user,
+      packagePolicy,
+      esClient: esClient.asCurrentUser,
+    });
+
+    logger.info('Finish updating rules');
+
+    return updatedPolicy;
+  } catch (e) {
+    logger.error('Failed updating rules in package policy vars');
+    logger.error(e);
+
+    logger.info('Rollback to previous saved-objects rules');
+    await soClient.bulkUpdate(currentRules);
+    logger.info('Finish rollback');
+
+    throw e;
+  }
+};
+
+/** @internal */
+export const updateRulesConfigurationHandler: CspRequestHandler<
+  void,
+  void,
+  UpdateRulesConfigBodySchema
+> = async (context, request, response) => {
+  const cspContext = await context.csp;
+
+  if (!(await context.fleet).authz.fleet.all) {
+    return response.forbidden();
+  }
+
+  try {
+    const updatedPackagePolicy = await updateRulesById(cspContext, request.body);
+    return response.ok({ body: updatedPackagePolicy });
+  } catch (e) {
+    const error = transformError(e);
+    cspContext.logger.error(
+      `Failed to update rules configuration on package policy - ${error.message}`
+    );
+
+    return response.customError({
+      body: { message: error.message },
+      statusCode: error.statusCode,
+    });
+  }
+};
+
 export const defineUpdateRulesConfigRoute = (router: CspRouter): void =>
   router.post(
```

Comment:
this is true, but note it's referring to a single resource and we're updating a bunch of them here.

i think it'd make more sense to use `PUT`/`PATCH` if the route for updating a resource was smth like `/api/csp/rules/:packagePolicyId/:policyId/:ruleId`

and the `request.body` would include the real new resource (full/partial) 
because then we're actually updating a specific resource. 

what we're doing now is different, we're sending `package_policy_id` and `rules[]` to `/api/csp/update_rules_config` which handles *all* the resources update, in bulk. 

which is to say, passing `PUT/PATCH` with `package_policy_id` and `rules[]` doesn't seem like it adheres to REST as we're not sending the resource, we're sending instructions on which resources to update and how. 

wdyt?